apis/checkpoint/finalized_state.yaml

```diff
@@ -0,0 +1,44 @@
+get:
+  operationId: "getFinalizedCheckpointState"
+  summary: "Get full BeaconState object for finalized checkpoint state"
+  description: |
+     Returns full BeaconState object for a finalized checkpoint state from the WS period.
+     Depending on `Accept` header it can be returned either as json or as bytes serialized by SSZ
+  tags:
+    - Checkpoint
+  responses:
+    "200":
+      description: Success
+      headers:
+        Eth-Consensus-Version:
+          $ref: '../../beacon-node-oapi.yaml#/components/headers/Eth-Consensus-Version'
+      content:
+        application/json:
+          schema:
+            title: GetStateResponse
+            type: object
+            properties:
+              version:
+                type: string
+                enum: [ phase0, altair, bellatrix ]
+                example: "phase0"
```

Comment:
This doesn't include the `is_optimistic` flag the existing getState has.  Do we want to add that or should this endpoint simply refuse to return a state if it's optimistically syncing? We probably should be clear about that whichever way we go.

Also worth noting that the state being finalized would be affected by the chain head being optimistic even if the state returned itself is fully validated (because the optimistic head may have updated finalisation).

Reply:
IMO, we should be conservative on that and respond 404 if the node is optimistic. I'll add a respective note